src/engine/sparqlExpressions/PrefilterExpressionIndex.cpp

```diff
@@ -217,22 +230,22 @@ RelationalExpression<Comparison>::logicalComplement() const {
   switch (Comparison) {
     case LT:
       // Complement X < Y: X >= Y
-      return std::make_unique<GreaterEqualExpression>(referenceValue_);
+      return std::make_unique<GreaterEqualExpression>(rightSideReferenceValue_);
```

Comment:
You can also implement this function using a `ConstexprMap` like below, saving much code, and making the relevant part more visible.